modules/samples/authenticators/components/org.wso2.carbon.identity.sample.extension.authenticators/src/main/java/org/wso2/carbon/identity/sample/extension/auth/SampleFingerprintAuthenticator.java

```diff
@@ -0,0 +1,135 @@
+/*
+ * Copyright (c) 2018, WSO2 Inc. (http://www.wso2.org) All Rights Reserved.
+ *
+ * WSO2 Inc. licenses this file to you under the Apache License,
+ * Version 2.0 (the "License"); you may not use this file except
+ * in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+
+package org.wso2.carbon.identity.sample.extension.auth;
+
+import org.wso2.carbon.identity.application.authentication.framework.context.AuthenticationContext;
+import org.wso2.carbon.identity.application.authentication.framework.exception.AuthenticationFailedException;
+import org.wso2.carbon.identity.application.authentication.framework.util.FrameworkConstants;
+import org.wso2.carbon.identity.application.authentication.framework.util.FrameworkUtils;
+import org.wso2.carbon.identity.application.common.model.Property;
+import org.wso2.carbon.identity.core.util.IdentityUtil;
+
+import javax.servlet.http.HttpServletRequest;
+import javax.servlet.http.HttpServletResponse;
+import java.io.IOException;
+import java.net.URLEncoder;
+import java.nio.charset.StandardCharsets;
+import java.util.ArrayList;
+import java.util.List;
+
+/**
+ * Sample Fingerprint Authenticator.
+ */
+public class SampleFingerprintAuthenticator extends AbstractSampleAuthenticator {
+
+    private static final long serialVersionUID = 6439291340285653402L;
+    private static final String FINGERPRINT_APP_URL = "FingerprintAppUrl";
+
+    @Override
+    public boolean canHandle(HttpServletRequest request) {
+
+        return true;
+    }
+
+    @Override
+    protected String getPageUrlProperty() {
+
+        return FINGERPRINT_APP_URL;
+    }
+
+    @Override
+    public String getContextIdentifier(HttpServletRequest request) {
+
+        String identifier = request.getParameter("sessionDataKey");
+        return identifier;
+    }
+
+    @Override
+    public String getName() {
+
+        return "SampleFingerprintAuthenticator";
+    }
+
+    @Override
+    public String getFriendlyName() {
+
+        return "Sample Fingerprint Authenticator";
+    }
+
+    @Override
+    protected void processAuthenticationResponse(HttpServletRequest httpServletRequest,
+                                                 HttpServletResponse httpServletResponse,
+                                                 AuthenticationContext authenticationContext)
+            throws AuthenticationFailedException {
```

Comment:
I think we need to set a subject here. There should be a resulting subject of an authenticaion. If not I think we need to fix it.